plugins/obs-qsv11/QSV_Encoder.cpp

```diff
@@ -144,27 +149,71 @@ bool prefer_current_or_igpu_enc(int *iGPUIndex)
 			    (AdapterDesc.AdapterLuid.LowPart == luid.LowPart) &&
 			    (AdapterDesc.AdapterLuid.HighPart ==
 			     luid.HighPart)) {
-				hasCurrent = true;
-				*iGPUIndex = adapterIndex;
-				break;
+				adapter_info->hasCurrent = true;
+				adapter_info->currentGPUIndex = adapterIndex;
 			}
 
 			if (AdapterDesc.DedicatedVideoMemory <=
 			    512 * 1024 * 1024) {
-				hasIGPU = true;
-				if (iGPUIndex != NULL) {
-					*iGPUIndex = adapterIndex;
-				}
+				adapter_info->hasIGPU = true;
+				adapter_info->iGPUIndex = adapterIndex;
+				adapter_info->iGPUAV1 =
+					qsv_av1_supported(MFX_IMPL_HARDWARE2);
 			} else {
-				hasDGPU = true;
+				adapter_info->hasDGPU = true;
+				adapter_info->dGPUIndex = adapterIndex;
+				adapter_info->dGPUAV1 =
+					qsv_av1_supported(MFX_IMPL_HARDWARE);
```

Comment:
Respectfully, this code upsets me a lot. You have to initialize an AV1 encoder just to use an H264 encoder. This adds many seconds of initialization time to not just the AV1 encoder, but sadly, to the H264 encoder as well.

This is a terrible situation that no one should find acceptable.

Reply:
I'm going to merge the AV1 encoder without this diff for beta. The people over there at Intel have to figure out a better way to handle this situation. There's about 5 weeks for testing before final release. Something else needs to be figured out before then.

I will not accept this particular change that requires spawning multiple encoder sessions per singular encode initialization.

Reply:
I'm sorry, I just have a lot of stress lately and I didn't mean to be like this. In the future though, just remember to keep in mind initialization times.

The way this should be handled is by querying support on startup. If it's unavoidable to test codec support except by spawning and encoder, then it should probably be done on startup, not when initializing an encoder.

Again, I apologize for my harshness. I am just under a lot of stress right now and you don't deserve that.